### PR TITLE
Add support to print information about newly upgraded Jenkins

### PR DESCRIPTION
**Pull request by jxr98**
**Make sure that you've checked the boxes below before you submit PR:**

### Always

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Written well with PR title, we generate the [release notes](https://github.com/jenkins-zh/jenkins-cli/releases) base on that

### For the bug fixes or features only

- [ ] [Quality Gate Passed](https://sonarcloud.io/dashboard?id=jenkins-zh_jenkins-cli). Change this URL to your PR.
- [ ] The coverage is xxx on the new lines
- [ ] I've tested it by manual in the following platform
  - [ ] MacOS
  - [ ] Linux
  - [x] Windows
- [x] Unit Test covered
- [ ] e2e Test covered

### CODE REVIEW

File: app/cmd/center_list_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	})
+
+	Context("basic cases", func() {
+		It("no jenkins version information in the list", func() {
+			data, err := GenerateSampleConfig()
+			Expect(err).To(BeNil())
+			err = ioutil.WriteFile(rootOptions.ConfigFile, data, 0664)
+			Expect(err).To(BeNil())
+
+			resp, _ := http.Get(LTSURL)
```
Comment (LinuxSuRen) [inline app/cmd/center_list_test.go:44]: Any real network request should not be here. We cannot guarantee all environments have internet access.

File: app/cmd/center_list.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jenkins-zh/jenkins-cli/app/i18n"
+	"github.com/jenkins-zh/jenkins-cli/client"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+const LTSURL = "https://www.jenkins.io/changelog-stable/rss.xml"
```
[the remark this answers is not in the document]
Comment (jxr98) [reply app/cmd/center_list.go:14]: OK, I have changed this part.

File: app/cmd/center_list_test.go (hunk 1 above)
Comment (yJunS) [inline app/cmd/center_list_test.go:44]: Where is the `LTSURL` value defined??

File: app/cmd/center_list_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+			bytes, _ := ioutil.ReadAll(resp.Body)
+			var centerListOption CenterListOption
+			xml.Unmarshal(bytes, &centerListOption)
+			theNewestJenkinsVersion := centerListOption.Channel.Items[0].Title[8:]
+			temp, _ := printChangelog(LTSURL, theNewestJenkinsVersion)
```
Comment (yJunS) [inline app/cmd/center_list_test.go:49]: Same as above

**Review by yJunS — CHANGES_REQUESTED** [review]

File: app/cmd/center_list_test.go (hunk 1 above)
Comment (jxr98) [reply app/cmd/center_list_test.go:44]: In the center_list.go. The LTSURL is a constant and I changed the name from LTSURL to LtsUrl. But I haven't changed the test file. I will have it done in a few days

Comment (jxr98) [reply app/cmd/center_list_test.go:44]: OK. I am going to change this.

File: app/cmd/center_list_test.go [hunk 4]
```diff
@@ -1,6 +1,7 @@
 package cmd
 
 import (
+	bytes2 "bytes"
```
Comment (LinuxSuRen) [inline app/cmd/center_list_test.go:4]: Why do you give it an alias `bytes2`?

File: app/cmd/center_list_test.go [hunk 5]
```diff
@@ -35,21 +36,87 @@ var _ = Describe("center list command", func() {
...
+&lt;li&gt; Bug: Fix form submission for some specific form validation cases (regression in 2.289). &lt;/li&gt;
+&lt;li&gt; Bug: Wrap the build name in the build results list if it is too long. &lt;/li&gt;
+ &lt;/ul&gt; </description>
+<pubDate> Wed, 2 Jun 2021 00:00:00 +0000 </pubDate>
+</item>
+</channel>
+</rss>`
+}
+
+func resultOneVersionData() string {
```
Comment (LinuxSuRen) [inline app/cmd/center_list_test.go:106]: I don't block here. But a better way might be putting these strings into a file. Golang allows us to embed files into the binary file.

## Refs
* https://golang.org/pkg/embed/
* https://blog.jetbrains.com/go/2021/06/09/how-to-use-go-embed-in-go-1-16/

File: app/cmd/center_list.go [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+//URL of stable Jenkins RSS
+const LtsURL = "https://www.jenkins.io/changelog-stable/rss.xml"
+//URL of stable Jenkins RSS
+const WeeklyYURL = "https://www.jenkins.io/changelog/rss.xml"
+//the width of the description column
+const WidthOfDescription = 60
+//the number of lines to be printed in description column
+const NumberOfLinesOfDescription = 10
+//ASCII of line feed
+const AsciiOfLineFeed = 10
```
[the remark this answers is not in the document]
Comment (LinuxSuRen) [reply app/cmd/center_list.go:27]: @jxr98 Please check the tips from hound.

File: app/cmd/center_list.go [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+//ASCII of space
+const AsciiOfSpace = 32
```
[the remark this answers is not in the document]
Comment (LinuxSuRen) [reply app/cmd/center_list.go:30]: @jxr98 Please check the tips from hound.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
Hi @jxr98 , thanks for your contribution. I found some small issues. Please take a look at it. And please don't forget to check the CI status from GitHub. The status of Lint was failed.

By the way, the e2e tests have a big chance to fail. So I suggest that we can skip them for now. See also https://github.com/jenkins-zh/jenkins-cli/blob/master/.github/workflows/e2e-test.yaml

**Comment by jxr98** [comment]
The reason why the e2e test failed this time was the go version used by this project was 1.15, but "embed" was added in go 1.16. I am gonna find a solution.

**Comment by LinuxSuRen** [comment]
> failed

We can use go 1.16

**Comment by yJunS** [comment]
> The reason why the e2e test failed this time was the go version used by this project was 1.15, but "embed" was added in go 1.16. I am gonna find a solution.

If you are sure that go 1.16 version can be tested normally in e2e, we can update the test dependency version to 1.16.

**Comment by yJunS** [comment]
#599  I already pull request to update e2e go version

**Comment by jxr98** [comment]
Hello @LinuxSuRen! Sorry to bother you. The e2e test failed again after I pulled from origin and resolved the conflict. The TestPowerShellCompletion failed and I have no idea what the reason is. Could you please give me a little hint on it? Thanks so much.

**Comment by LinuxSuRen** [comment]
> Hello @LinuxSuRen! Sorry to bother you. The e2e test failed again after I pulled from origin and resolved the conflict. The TestPowerShellCompletion failed and I have no idea what the reason is. Could you please give me a little hint on it? Thanks so much.

Let's forget about the e2e tests for now.

**Comment by jxr98** [comment]
> > Hello @LinuxSuRen! Sorry to bother you. The e2e test failed again after I pulled from origin and resolved the conflict. The TestPowerShellCompletion failed and I have no idea what the reason is. Could you please give me a little hint on it? Thanks so much.
> 
> Let's forget about the e2e tests for now.

All right.

**Comment by LinuxSuRen** [comment]
It would be better if you can tell us how to test this PR.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
I feel confused about the following output. Maybe for many users, they just don't know what they should do next. So, it would be great to have more information about your CLI:
```
gitpod /workspace/jenkins-cli $ ./jcli center list
Error: Get "http://localhost:8080/jenkins/api/json": dial tcp [::1]:8080: connect: connection refused
```

**Comment by jxr98** [comment]
> I feel confused about the following output. Maybe for many users, they just don't know what they should do next. So, it would be great to have more information about your CLI:
> 
> ```
> gitpod /workspace/jenkins-cli $ ./jcli center list
> Error: Get "http://localhost:8080/jenkins/api/json": dial tcp [::1]:8080: connect: connection refused
> ```

I will figure this out tomorrow.

**Pull request merged**